File: 92/b/b.py
import sys
from itertools import permutations

n, k = map(int, sys.stdin.readline().split())

x = []
for i in xrange(n):
    x.append(sys.stdin.readline().strip())


def permute(n, perm):
    ret = []
    for i in perm:
        ret.append(n[i])
    return int(''.join(ret))

ret = float('inf')
for p in permutations(range(k)):
    xmax = -1
    xmin = float('inf')
    for n in x:
        xmax = max(xmax, permute(n, p))
        xmin = min(xmin, permute(n, p))
    ret = min(ret, xmax-xmin)

print "["
for i in xrange(1, 9):
    perms = []
    for p in permutations(range(i)):
        perms.append(p)
    print perms, ","
print "]"

File: 92/b/b.cpp
#include <cstdio>
#include <algorithm>
#include <vector>
#include <fstream>
#include <sstream>
#include <iostream>
#include <string>
#include <map>
#include <set>
#include <queue>
#include <cstring>

using namespace std;

int permute(string& s, vector<int>& perm) {
  string ret = "";
  for (int i = 0; i < perm.size(); ++i) {
    ret += s[perm[i]];
  }
  return atoi(ret.c_str());
}

int main() {
  int n, k;
  cin >> n >> k;

  vector<string> x;
  for (int i = 0; i < n; ++i) {
    string s;
    cin >> s;
    x.push_back(s);
  }

  vector<int> perm;
  for (int i = 0; i < k; ++i)
    perm.push_back(i);

  int ret = 99999999;
  do {
    int xmax = -1;
    int xmin = 99999999;
    for (int i = 0; i < x.size(); ++i) {
      xmax = max(xmax, permute(x[i], perm));
      xmin = min(xmin, permute(x[i], perm));
    }
    ret = min(ret, xmax-xmin);
  } while (next_permutation(perm.begin(), perm.end()));

  cout << ret << endl;
  return 0;
}
